packages/composable-controller/src/ComposableController.ts

```diff
@@ -28,31 +31,69 @@ export type ControllerList = ControllerInstance[];
  * @param controller - Controller instance to check
  * @returns True if the controller is an instance of BaseControllerV1
  */
-function isBaseControllerV1(
+export function isBaseControllerV1(
   controller: ControllerInstance,
-  // TODO: Replace `any` with type
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-): controller is BaseControllerV1<any, any> {
-  return controller instanceof BaseControllerV1;
+): controller is BaseControllerV1<
+  BaseConfig & Record<string, unknown>,
+  BaseState & Record<string, unknown>
```

Comment:
The BaseControllerV1 `childState` in `#updateChildController` is correctly inferred to this state type.